### DISCUSSION TITLE
debug GetStartedApp with error of  file not found

### DISCUSSION THREAD

**Discussion by cd-willi** [AvaloniaUI/Avalonia#14069] [Q&A] [answered]
Hi, folks. I am new hand. I encounter a problem while debugging TestStartedApp using Visual Studio 2022 Communite:
The type initializer for 'Avalonia.Diagnostics.Conventions' threw an exception.
  Source=Avalonia.Diagnostics
  StackTrace:
   at Avalonia.Diagnostics.Conventions.get_DefaultScreenshotHandler()
   at Avalonia.Diagnostics.DevToolsOptions..ctor()
   at Avalonia.DevToolsExtensions.AttachDevTools(TopLevel root)
   at GetStartedApp.Views.MainWindow.InitializeComponent(Boolean loadXaml, Boolean attachDevTools) in C:\projects\2023\c#\avalonia\GetStartedApp\GetStartedApp\Avalonia.Generators\Avalonia.Generators.NameGenerator.AvaloniaNameSourceGenerator\GetStartedApp.Views.MainWindow.g.cs:line 29
   at GetStartedApp.Views.MainWindow..ctor() in C:\projects\2023\c#\avalonia\GetStartedApp\GetStartedApp\Views\MainWindow.axaml.cs:line 9
   at GetStartedApp.App.OnFrameworkInitializationCompleted() in C:\projects\2023\c#\avalonia\GetStartedApp\GetStartedApp\App.axaml.cs:line 21
   at Avalonia.AppBuilder.SetupUnsafe()
   at Avalonia.AppBuilder.Setup()
   at Avalonia.AppBuilder.SetupWithLifetime(IApplicationLifetime lifetime)
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder builder, String[] args, ShutdownMode shutdownMode)
   at GetStartedApp.Desktop.Program.Main(String[] args) in C:\projects\2023\c#\avalonia\GetStartedApp\GetStartedApp.Desktop\Program.cs:line 14

  This exception was originally thrown at this call stack:
    [External Code]

Inner Exception 1:
FileNotFoundException: Could not find file 'C:\Users\myname\OneDrive\image\Screenshots'.

**Comment by cd-willi** [comment]
Why the first app 'Welcome to Avalonia!' trys to access a subdir of my onedrive folder which actually not exists.

**Reply by cd-willi** [reply]
My first date with Avalonia was pinned with doubt of security concern?

**Comment by jp2masa** [comment, marked as answer]
You can look at the source, here:

https://github.com/AvaloniaUI/Avalonia/blob/4a30a546fa17c95a06ec0da76c0b67707088a986/src/Avalonia.Diagnostics/Diagnostics/Conventions.cs#L8-L17

So, that path is returned by `Environment.GetFolderPath(Environment.SpecialFolder.MyPictures)`. The [.NET docs](https://learn.microsoft.com/en-us/dotnet/api/system.environment.getfolderpath?view=net-8.0#system-environment-getfolderpath(system-environment-specialfolder)) specify that the return value is:

> The path to the specified system special folder, if that folder physically exists on your computer; otherwise, an empty string ("").
>
> A folder will not physically exist if the operating system did not create it, the existing folder was deleted, or the folder is a virtual directory, such as My Computer, which does not correspond to a physical path.

So, that folder should exist.

**Reply by thevortexcloud** [reply]
That sounds like a possible bug to me. It's entirely possible you will end up with "\Screenshots" as the combined path. In which case you would likely get undefined behaviour.

On a side note, the exception is `FileNotFoundException`. Which is either incorrect (as `Screenshots` is the directory) or something weird is happening.

**Reply by cd-willi** [reply]
THanks, now I know where the problem comes. The program launchs successully in debug mode after I created the directory manually.

However, as hello-world template for Avalonia, I think there is no need to make things complicated. Would this Avalonia's hello-world program try to make snapshot for such a simple UI?

**Reply by thevortexcloud** [reply]
Which sample are you trying to run? As far as I am aware the default app created by the template does not take any screenshots.

**Reply by cd-willi** [reply]
I follow the guide from https://docs.avaloniaui.net/docs/get-started/test-drive/create-a-project,   and create the app using the command: dotnet new avalonia.app -o GetStartedApp.

**Reply by cd-willi** [reply]
My OS is Windows 11 version 22H2. I have installed Visual Studio 2022 17.8.0 and extensions Avalonia Toolkit, Avalonia Template Studio and Avalonia for Visual Studio 2022. I created this first Avalonia app following guide and didn't make any editing, and then encountered the problem described above while running it in debug mode.

**Reply by thevortexcloud** [reply]
Sounds like it's just an issue with the diagnostics package. By default it only runs on debug builds. You could of course just remove the package to disable it entirely. But you do lose access to useful things like the inspector if you do that.

**Reply by cd-willi** [reply]
@jp2masa， this line of code Directory.CreateDirectory(dir) might not work correctly, since I solved the problem by manually creating the subdir.
